### PR TITLE
CI: general improvements

### PR DESCRIPTION
**Pull request by Fraccaman**
- [x] parallel e2e
- [x] update mold to 1.7.0
- [x] added timing report to gh artifacts
- [x] added coverage report to gh artifacts

### CODE REVIEW

File: .github/workflows/build-and-test-bridge.yml [hunk 1]
```diff
@@ -66,13 +65,49 @@ jobs:
...
+        with:
+          name: wasm-${{ github.event.pull_request.head.sha|| github.sha }}
+          path: ./wasm
+        env:
+          RUSTFLAGS: "-C linker=clang -C link-arg=-fuse-ld=/usr/local/bin/mold"
       - name: Test Wasm
         run: make test-wasm
       - name: Check wasm up-to-date
-        run: cmp -- wasm/checksums.json wasm/original-checksums.json
+        run: cmp -- wasm/checksums.json wasm/original-checksums.json || true
```
Comment (tzemanovic) [inline .github/workflows/build-and-test-bridge.yml:107]: will this still catch `if: failure()` below?

File: .github/workflows/scripts/e2e.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (tzemanovic) [inline .github/workflows/scripts/e2e.json:1]: what's the numbers here? I see it's being used as 'total_time' below, but I don't follow why and where it comes from

File: .github/workflows/build-and-test-bridge.yml (hunk 1 above)
Comment (Fraccaman) [reply .github/workflows/build-and-test-bridge.yml:107]: oh my bad here, the bridge team asked me to let the CI pass when if the checksum is not up-to-date. But this is the wrong workflow file, ill fix this! Thanks

File: .github/workflows/scripts/e2e.json (hunk 2 above)
Comment (Fraccaman) [reply .github/workflows/scripts/e2e.json:1]: they are taken from `cargo test` output, an example here https://github.com/anoma/namada/actions/runs/3469076888/jobs/5795628173

Comment (Fraccaman) [reply .github/workflows/scripts/e2e.json:1]: this means that we will have to update this file everytime we add a new e2e test, which I thinks its reasonable and can be automated later

**Review by tzemanovic — APPROVED** [review]

**Pull request merged**